src/fairseq2/data/data_pipeline.py

```diff
@@ -94,24 +94,36 @@ def zip(
             """
 
         @staticmethod
-        def round_robin(pipelines: Sequence["DataPipeline"]) -> "DataPipelineBuilder":
+        def round_robin(
+            pipelines: Sequence["DataPipeline"],
+            stop_at_shortest: bool = False,
+        ) -> "DataPipelineBuilder":
             """Extract examples from ``pipelines`` in round robin.
 
             :param pipelines:
                 The data pipelines to round robin.
+            :param stop_at_shortest:
+                Flag to stop round_robin when first pipeline reaches its end.
```

Comment:
Nit: As a convention for flag parameters, I use the template "If ``True``, blab bla". In this case, that would look like: "If ``True``, stops round robin when first pipeline reaches its end."